Automattic/pocket-casts-android
# Add episode screen and associated functionality to watch app

## Description
Adds the episode screen to the watch app along with the associated functionality.

The `EpisodeViewModel` implementation very closely follows the logic in the `EpisodeFragment` and `EpisodeFragmentViewModel`.

## Testing Instructions

Verify that you can access the episode screen in each of the following ways:

* Tapping on the episode/podcast name in the Now Playing screen
* Tapping on an episode in the Up Next queue
* Tapping on an episode on the podcast screen

From the episode screen, you should see the following UI:

* The Top of the screen has the podcast name, followed by the podcast title
* The download, play, and, queue buttons are tinted based on the podcast
* Below the play button is the episode date, amount of time left, and download size of the episode
* Below that is a list with the options: archive, star, mark as played, and go to podcast

From the episode screen, check the following functionality:

* Tapping the download button should have the following actions (with the button UI updating appropriately for each state):
  * if not downloaded, start a download
  * if a download is in progress, cancel the download
  * if the episode is downloaded, present a confirmation dialog for deleting the download. Confirming on that dialog should delete the download and briefly present a notification screen notifying the user that the download was removed (tapping on the notification screen will dismiss the notification even earlier than that 2 second timeout).
* Tapping the play button should start playing the episode (or pause it if it is already playing. The UI should reflect whether the button is a play or a pause button.
* Tapping the add-to-queue button should:
  * If the episode is in the Up Next queue, remove the episode from the queue and briefly present a notification dialog advising the user that the episode was removed from the queue.
  * If the episode is not in the Up Next queue, and...
    * If the Up Next queue is empty, add the episode to the Up Next queue
    * If the Up Next queue is not empty, present a screen asking where to add the episode to the queue.
 * Tapping the archive/unarchive, star/unstar,  mark as played/unplayed buttons should update the episode state appropriately, and the button's UI should update to reflect the new state.
 * Tapping the go-to-podcast button should take you to the podcast screen

## Screenshots or Screencast 

![image](https://user-images.githubusercontent.com/4656348/223600038-22747b7f-0303-4e2f-a5b9-30fa4a790167.png)


## Checklist
 
#### I have tested any UI changes...
